### ISSUE TITLE
`pre-commit` Fails to Recognize `per-file-ignores`

### ISSUE THREAD

**Issue by adam-grant-hendry** [astral-sh/ruff#14640]
### System
OS: Windows 10 22H2
Python: 3.9.12
Ruff: 0.8.0
Pre-Commit; 2.18.1
VSCode: 1.95.3
Ruff Extension: v2024.56.0

### `.pre-commit-config.yml` (**Relevant Portion**)
```yml
fail_fast: true
repos:
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.8.0
    hooks:
      - id: ruff
      - id: ruff-format
```

### `pyproject.toml` (**Relevant Portion**)
```toml
[tool.ruff.lint.per-file-ignores]
"__init__.py" = ["F401"]  # "Imported but unused: happens with packages
```

### Error
When running either of:

```
uvx pre-commit run --all
```

```
pre-commit run --all
```

The following error occurs:

```
ruff failed
  Cause: Failed to parse C:\Users\ahendry\Code\qtpygraph\pyproject.toml
  Cause: TOML parse error at line 177, column 12
    |
177 | [tool.ruff.lint.per-file-ignores]
    |            ^^^^
unknown field `ignore`, expected one of `convention`, `ignore-decorators`, `property-decorators`
```

Interestingly, the `ruff` VSCode extension complains that it runs into an error sporadically, but the Output window only shows `info` logs, which do not seem to indicate any errors have occurred:

![Image](https://github.com/user-attachments/assets/09dad861-eff4-4abd-8f23-26e100d6c9fe)

![Image](https://github.com/user-attachments/assets/3a73ea25-a19d-47b4-8393-ae963de74022)

I have not configured any settings for `ruff` in my `settings.json`.

Perhaps related, but the `Even Better TOML` VSCode extension also seems to flag an issue:

![Image](https://github.com/user-attachments/assets/f7656235-e0d1-44d1-92f8-098ce7d42288)

I'm not sure what is causing this. I feel like I'm missing something obvious...? Any help would be much appreciated. Please let me know if you need more information.

### Possibly Relevant Issues
#12058 [#359](https://github.com/koxudaxi/ruff-pycharm-plugin/issues/359#issuecomment-2192785432)

**Comment by MichaReiser** [comment]
Hi. The error message suggests that there's an error in your pyproject.toml file. I admit, the error message isn't very helpful because it doesn't point you to the correct line number. 

I think the error is that the `ignore` setting on line `194..197` should be under the `[tool.ruff.lint]` table and not in `tool.ruff.lint.pydocstyle`.

**Comment by MichaReiser** [comment]
I'm glad to hear that! 

Yes, we're tracking the improvement of the error message in https://github.com/astral-sh/ruff/issues/9719

**Comment by adam-grant-hendry** [comment]
Beautiful, thanks! Closing since this is user error and not a bug.
